master/getting-started/mesos/tutorials/connecting-tasks.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Connecting to Tasks
+---
+
+Calico assigns each mesos task a uniquely addressable IP address.
+In testing, it is easy to test connectivity by making requests directly to this
+IP.
+
+However, when designing a multi-tier application, it is not usually sensible to
+hardcode applications to statically connect to these addresses directly, as they are dynamic
+and often change as the tasks die and are reinstantiated across the cluster.
+
+Instead, it is advisable to program the applications to connect to one another
+using DNS entries which dynamically resolve to the applications' current addresses.
+
+In Mesos, there are several Service Discovery solutions which will can perform
+the resolution to the task's container IP.
+
+### Mesos-DNS
+
+Mesos-DNS is an open source DNS service for Mesos which gathers task information
+from the Mesos Master and uses it to serve DNS entries which resolve to a task's IP.
+Mesos-DNS provides the following DNS entry for tasks launched by Marathon:
+
+```
+<task-name>.marathon.mesos
+```
+
+By default, Mesos-DNS will return the IP address of the _Agent_ that is running the task.
+It can be configure to instead return the task's Container IP by modifying its configuration
+to preferentially resolve to a task's `netinfo` for the `IPSources` field, as follows:
+
+```
+{
+  "IPSources": ["netinfo", "host"]
+}
+```
+
+For example, a Calico-networked task named `nginx` scaled to 3 instances will
+respond to DNS lookups with the Calico IPs:
+
+```
+[vagrant@calico-01 ~]$ nslookup nginx.marathon.mesos
+Server:         172.24.197.101
+Address:        172.24.197.101#53
+
+Name:   nginx.marathon.mesos
+Address: 192.168.84.192
+Name:   nginx.marathon.mesos
+Address: 192.168.50.65
+Name:   nginx.marathon.mesos
+Address: 192.168.50.66
+```
+
+>Warning: Making this change in DC/OS may break components which rely on Mesos-DNS
```

Comment:
Maybe also have a warning at the top of the section and state that this is not valid for DC/OS.